internal/config/drive/drive.go

```diff
@@ -65,8 +69,9 @@ func LookupConfig(kvs config.KVS) (cfg Config, err error) {
 	if err = config.CheckValidKeys(config.DriveSubSys, kvs, DefaultKVS); err != nil {
 		return cfg, err
 	}
+
 	// if not set. Get default value from environment
-	d := kvs.GetWithDefault(MaxTimeout, DefaultKVS)
+	d := env.Get(EnvMaxDriveTimeout, kvs.GetWithDefault(MaxTimeout, DefaultKVS))
```

Comment:
found this bug which is missing we didn't honor MINIO_DRIVE_MAX_TIMEOUT by mistake